js/app/controllers/calcontroller.js

```diff
@@ -140,167 +140,177 @@ app.controller('CalController', ['$scope', 'Calendar', 'CalendarService', 'VEven
 				$scope.$apply();
 			});
 		} else {
-			$scope.calendarsPromise = CalendarService.getPublicCalendar(constants.publicSharingToken).then(function(calendar) {
-				$scope.calendars = [calendar];
+			$scope.calendars = [];
+			var tokens = new Set(constants.publicSharingToken.split("."));
+			var promises = [];
+			tokens.forEach(function(token){
+				promises.push(CalendarService.getPublicCalendar(token).then(function(calendar) {
+					$scope.calendars.push(calendar);
+				}).catch((reason => {
+					OC.Notification.showTemporary(t('calendar', 'Calendar does not exist'));
```

Comment:
_OC.Notification_ does not work in calendars public sharing window. I think this is an issue somewhere else in the calendar app.

Reply:
Ok, I will check.